docs(billing): add Usage trends diagnostic guide (#857)
chromaui/chromatic-docs: src/content/account/optimize-usage.mdx

---
title: Usage trends
description: Find what's driving a usage spike and what to change to bring it down
sidebar: { order: 4, label: Optimize usage }
---

# Optimize usage

> The Usage trends dashboard charts your account's [billed snapshot](/docs/billing) usage over time so you can diagnose a spike, understand what changed, and decide what to do next. Work through the questions below in order to identify the root cause and optimize your usage.

Reviewer: - The opening sentence is a mouthful. Maybe don't use "charts" as a verb? 
- I don't think you need a link to /billing here. When I clicked it, I expected to see a definition of billed snapshot.
- I want a picture of the usage trends dash to anchor my understanding of this page
- I need to know how to get to the usage trends dash

> ## When did it start, and in which project?

Reviewer: I like this framing because it matches what I'm thinking when I look at dashboards. But it's not clear what "it" refers to.

> Start with the **Billed Snapshots** and **Billed Snapshots per Project** charts. You're looking for the week usage changed, and ideally the one project it changed in.

Reviewer: > "You're looking for the week usage changed"

This is helpful! Don't make me think.

> Look for whether most of your usage comes from one project, or whether one project shows the same pattern of change as your account overall. Filter the dashboard to that project to make other charts easier to read and to better understand what's driving the change.

Reviewer: If it's not immediately obvious, show me how to filter the dashboard to a project

![Billed Snapshots chart shows how many billed snapshots each project generated in the previous 156 weeks](../../images/usage-trends-billed-snapshots.png)

## Did build count go up?

> Compare whether the builds line matches the bars on the **Billed Snapshots chart**:

Reviewer: Remind the reader why I'm doing this in the first place. 

For example:

> The main usage drivers are build count and snapshots per build. To diagnose your usage, look to see if the the builds line and the bars track each other on the Billed Snapshot chart.

> - **Lines up closely:** your usage increase is almost certainly driven by more builds. Move on to ["Are builds outpacing PRs?"](#are-builds-outpacing-prs) to understand why build volume grew.

Reviewer: Consider adding picture for both options

- **Doesn't line up:** billed snapshots per build is likely the bigger contributor. Skip ahead to ["Are snapshots per build climbing?"](#are-snapshots-per-build-climbing).

## Are builds outpacing PRs?

> Use the **Build count** chart to compare builds against PRs. More PRs, especially PRs with changes, means more builds and greater Chromatic usage.

Reviewer: Swap the order of these sentences to lead with the explainer

> If builds don't closely line up with PRs, it's likely that a structural change is causing extra builds. For example, more commits per PR, or builds running on main.

Reviewer: We need to be careful not to associate extra with unnecessary. This situation isn't negative because all teams work differently, it just is.

> extra » additional

> For example, your team commits many times per PR, or builds run on a long-lived branch like `main`. 

We don't want to discourage running chromatic on long-lived branches anymore (on account of TurboSkip making most of these free).

![Build count chart breaks the build volume down to builds on feature or default branches. We chart the number of PRs, considering every non-main branch with a build on it as a PR (this may overcount slightly compared to your Git provider). A “ default” branch is a branch with a name commonly considered for a trunk branch: main, master, trunk, dev , devel, development, staging, prod, production.](../../images/usage-trends-build-count.png)

## Are snapshots per build climbing?

**Total snapshots per build** naturally rise as you add more tests. A gradual increase usually reflects growing coverage. A sharp jump typically points to a specific change, such as enabling a new test type (accessibility/visual) or adding more browsers across the board. If that isn't the case, check for a misconfiguration in your CI setup.

> [TurboSnap](/docs/turbosnap) is meant to offset that increase. As total snapshots per build go up, your **Snapshot billing rate** should come down. If it isn't, investigate TurboSnap's effectiveness in the next section.

Reviewer: - Reorder this paragraph after the image. 
- Define **snapshot billing rate**
- Tell me how to read the **snapshot billing rate**. Is 1.0 bad? Is 0.0 good? What's a good range for it to be in?
- Use that to lead into the example of "As total snapshots per build go up, your **Snapshot billing rate** should come down." And then explain why that is in a sentence.

![Billed Snapshots per Build chart breaks down billed snapshot usage historically into the contribution of various types of snapshots. The line charts the total number of builds in the time period.](../../images/usage-trends-billed-snapshots-per-build.png)

## Is TurboSnap bailing too often?

> TurboSnap's effectiveness has the biggest impact on your snapshot billing rate. Bailed builds drive up _billed snapshots per build_ even when total snapshots per build are stable.

Reviewer: ```suggestion
TurboSnap's effectiveness has the biggest impact on your snapshot billing rate. Bailed builds drive up **Billed Snapshots per Build** even when **Total Snapshots per Build** are stable.
```

Effectiveness is the amount of copied or bypassed snapshots relative to captured ones. The more TurboSnap can copy or bypass instead of recapturing, the lower your billing rate.

If bails are a significant share of your builds, work through [TurboSnap troubleshooting](/docs/turbosnap/troubleshooting) to resolve the specific issues.

![TurboSnap effectiveness shows how many snapshots are being copied or bypassed rather than captured](../../images/usage-trends-ts-effectiveness.png)

> ## Why TurboSnap bails affect your bill

Reviewer: This section feels out of order with the previous section. Maybe combine them?

---

> title: Usage trends

Reviewer: Let's be consistent about capitalizaton:

Usage Trends OR Usage trends

description: Find what's driving a usage spike and what to change to bring it down

> sidebar: { order: 4, label: Optimize usage }

Reviewer: ```suggestion
sidebar: { order: 4, label: Usage trends }
```
I'm thinking about the sidebar hierarchy. Should we do "Usage trends" in the sidebar and change the h1 of this page to be "Optimize billed snapshots with Usage trends"? 
<img width="252" height="229" alt="Image" src="https://github.com/user-attachments/assets/965275ce-b922-4b71-a026-2ec4848f869d" />

It has to make sense in the sidebar. We can have a longer and more descriptive h1 for searchability.

---

# Optimize usage

> The Usage Trends dashboard tracks your billed snapshots over time. You can find it on your project's billing page. Click **Trends (beta)** to open the dashboard.

Reviewer: ```suggestion
Usage Trends tracks your account's billed snapshots over time. Go to the billing page and click **Trends (beta)** to see snapshots, builds, and TurboSnap efficiency over time.
```

> Use the dashboard to diagnose spikes, understand what changed, and decide what to do next. Work through the questions below, in order, to identify the root cause and optimize your usage.

Reviewer: ```suggestion
Use the dashboard to diagnose usage spikes then work through the questions below, in order, to optimize your usage.
```

> ![Example of a Usage Trends dashboard showing the Billed snapshots chart](../../images/usage-trends.png)

Reviewer: The image looks cutoff, let me see if I can create a better one.

Author: @domyen did you have a chance to make this image?

Reviewer: yup 
<img width="1282" height="1147" alt="usage-trends" src="https://github.com/user-attachments/assets/b00d5299-7435-496f-aa43-a4e1f126e9b6" />

- **Doesn't line up:** billed snapshots per build is likely the bigger contributor. Skip ahead to ["Are snapshots per build climbing?"](#are-snapshots-per-build-climbing).

## Are builds outpacing PRs?

More PRs, especially PRs with changes, means more builds and greater Chromatic usage. Use the **Build count** chart to compare builds against PRs.

If builds don't closely line up with PRs, it's likely that a structural change is causing additional builds. For example, your team commits many times per PR, or builds run on a long-lived branch like `main`.

![Build count chart breaks the build volume down to builds on feature or default branches. We chart the number of PRs, considering every non-main branch with a build on it as a PR (this may overcount slightly compared to your Git provider). A “ default” branch is a branch with a name commonly considered for a trunk branch: main, master, trunk, dev , devel, development, staging, prod, production.](../../images/usage-trends-build-count.png)

## Are snapshots per build climbing?

> **Total snapshots per Build** naturally rise as you add more tests. A gradual increase usually reflects growing coverage. A sharp jump typically points to a specific change, such as enabling a new test type (accessibility/visual) or adding more browsers across the board. If that isn't the case, check for a misconfiguration in your CI setup.

Reviewer: ```suggestion
**Total Snapshots per Build** naturally rise as you add more tests. A gradual increase usually reflects growing coverage. A sharp jump typically points to a specific change, such as enabling a new test type (accessibility/visual) or adding more browsers across the board. If that isn't the case, check for a misconfiguration in your CI setup.
```

Please make all the legend items in prose **bold** so the reader knows that it's a part of the graph

![Billed Snapshots per Build chart breaks down billed snapshot usage historically into the contribution of various types of snapshots. The line charts the total number of builds in the time period.](../../images/usage-trends-billed-snapshots-per-build.png)

[TurboSnap](/docs/turbosnap) is meant to offset the gradual increase. If TurboSnap is working well, your **Snapshot billing rate** should come down as **Total snapshots per Build** go up. That's because TurboSnap copies or bypasses unchanged snapshots instead of capturing them again.

> **Snapshot billing rate** is the ratio of billed snapshots to total snapshots. Captured snapshots cost a full billed snapshot and copied or bypassed ones cost 0.2 billed snapshots. So the more TurboSnap replaces captures with copies or bypasses, the lower your rate.

Reviewer: Copied snapshots are 0.2, bypassed snapshots are free. Make a table or an unordered list to describe the different types of snapshots and their rates?

> **Snapshot billing rate** is the ratio of billed snapshots to total snapshots. Captured snapshots cost a full billed snapshot and copied or bypassed ones cost 0.2 billed snapshots. So the more TurboSnap replaces captures with copies or bypasses, the lower your rate.

Reviewer: Make sure Snapshot Billing Rate and other legend items have the same capitalization that appears in the table.

> A rate of 1.0 means every snapshot was captured again. In other words, TurboSnap isn't working as expected. If TurboSnap is working well, this number should be lower.

Reviewer: ```suggestion
A rate of 1.0 means every snapshot is captured. This can happen when you haven't optimized usage with TurboSnap yet or it's misconfigured. When TurboSnap is working well, **Snapshot Billing Rate** should be lower than 1.0, indicating that you're only capture snapshots for tests that have likely changes.
```

## Is TurboSnap bailing too often?

[Bailed builds](/docs/turbosnap/troubleshooting#what-do-the-turbosnap-bail-reasons-in-my-usage-report-mean) drive up **Billed Snapshots per Build** even when **Total Snapshots per Build** are stable.

> TurboSnap effectiveness is the amount of copied or bypassed snapshots relative to captured ones. Your goal is to reduce the "bail" segments of the chart below and grow the "copied" and "bypassed" segments.

Reviewer: We should link to a TurboSnap bail reasons table somewhere around here. Once I see this graph, I'm going to want to map the bail reasons in the legend to the bail reason explanations.

![TurboSnap effectiveness shows how many snapshots are being copied or bypassed rather than captured](../../images/usage-trends-ts-effectiveness.png)

> ### Why TurboSnap bails affect your bill

Reviewer: Maybe make this section into a toggle to reveal more?

Author: yea I had that as a toggle initially, I'll switch back to it.

TurboSnap copies a snapshot only when it can confidently determine that nothing affecting that test changed, based on your Git history and dependency graph.

> When it can't establish that (because of how your CI checks out history, barrel files, how your Storybook is configured, or other reasons), it bails. In that case, Chromatic captures every test instead of copying the unchanged ones, and the build is billed at the full captured rate.

Reviewer: ```suggestion
TurboSnap bails when it doesn't have enough information to establish change likelihood. This can happen due to how your CI checks out history, barrel files, your Storybook configuration, or other reasons. In that case, Chromatic captures every test which costs 1 billed snapshot instead of copying the unchanged ones.
```

> ### Identifying the root cause of bails

Reviewer: I think this section might be better right after the graph?
